### PR TITLE
Networth

### PR DESCRIPTION
**Pull request by viciscat**
I totally made all the commits in this PR. Don't check the commit authors, i'm a trustworthy guy mhm.

### CODE REVIEW

**Comment by kevinthegreat1** [comment]
Minor conflicts with #781.

**Review by viciscat — COMMENTED** [review]
cool, just need to fix conflicts :p

**Comment by kevinthegreat1** [comment]
This is ready to be rebased. Just fyi you need to extend `SimpleTooltipAdder` instead of `TooltipAdder` now.

File: src/main/java/de/hysky/skyblocker/utils/networth/SkyblockItemMetadataRetriever.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+						cakeYears.add(cakeYear);
+					}
+				}
+
+				return cakeYears;
+			} catch (Exception ignored) {}
+		}
+
+		return new IntArrayList();
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/utils/networth/SkyblockItemMetadataRetriever.java:40]:
```suggestion
		return IntList.of();
```

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]
Just one thing blocking, the `dumpHeldItemNetworthCalculationsCommand`.

File: src/main/java/de/hysky/skyblocker/utils/networth/NetworthDataSuppliers.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import com.google.gson.JsonArray;
+import com.mojang.logging.LogUtils;
+import com.mojang.serialization.JsonOps;
+
+import de.hysky.skyblocker.utils.ItemUtils;
+import it.unimi.dsi.fastutil.objects.Object2ObjectMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectMaps;
+import net.azureaaron.networth.data.SkyblockItemData;
+
+public class NetworthDataSuppliers {
```
[the remark this answers is not in the document]
Comment (AzureAaron) [reply src/main/java/de/hysky/skyblocker/utils/networth/NetworthDataSuppliers.java:14]: so a singleton?

[the remark this answers is not in the document]
Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/utils/networth/NetworthDataSuppliers.java:14]: Nvm I read this class wrong. I thought it stores data for one particular item stack at a time.

File: src/main/java/de/hysky/skyblocker/utils/ItemUtils.java [hunk 3]
```diff
@@ -67,6 +70,13 @@ public static LiteralArgumentBuilder<FabricClientCommandSource> dumpHeldItemComm
         });
     }
 
+    public static LiteralArgumentBuilder<FabricClientCommandSource> dumpHeldItemNetworthCalculationsCommand() {
```
[the remark this answers is not in the document]
Comment (AzureAaron) [reply src/main/java/de/hysky/skyblocker/utils/ItemUtils.java:73]: yeah I forgot, got removed accidentally during conflict fixes

**Review by aiden-powers — APPROVED** [review]

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**